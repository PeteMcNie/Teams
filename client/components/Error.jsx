import React from 'react'
import { connect } from 'react-redux'

const Error = (props) => {
  return (
    <div>{props.error}</div>
  )
}

function mapStateToProps (state) {
  return {
    error: state.error
  }
}

export default connect(mapStateToProps)(Error)
