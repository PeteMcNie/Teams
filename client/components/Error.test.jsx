import React from 'react'

import { renderWithRedux } from '../testing/utils'

import Error from './Error'

const initialState = {
  error: null
}

const newState = {
  error: 'error'
}

test('Error component is falsey when no error is thrown', async () => {
  const { asFragment } = renderWithRedux(<Error />, { initialState })
  expect(asFragment).toMatchSnapshot()
})

test('Error component is truthy when store changes', async () => {
  const { asFragment } = renderWithRedux(<Error />, { newState })
  expect(asFragment).toMatchSnapshot()
})
