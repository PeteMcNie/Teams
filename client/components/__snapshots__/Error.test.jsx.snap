// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component is falsey when no error is thrown 1`] = `[Function]`;

exports[`Error component is truthy when store changes 1`] = `[Function]`;
